### ISSUE TITLE
No license found

### ISSUE THREAD

**Issue by JacksonChen666** [Diggsey/five_words#1]
This project does not seem to have a license declared in an obvious way, or at all. A license should be added (as this project seems to be intended as open source) so that users can run, modify, study, distribute, and use the code.

https://choosealicense.com/no-permission/

**Comment by Diggsey** [comment]
>  this project seems be intended as open source

What makes you think that?

Please don't open spam issues.

![image](https://user-images.githubusercontent.com/451321/204847039-24431fdf-6222-4576-8298-28d1a70c369c.png)

**Comment by JacksonChen666** [comment]
> > this project seems be intended as open source
> 
> What makes you think that?
> 

You know, the code is available publicly. In a GitHub repo.

(Disclaimer: I am not a lawyer, this is not legal advice, and the information provided here might be inaccurate or outdated.)

Just making your code publicly available doesn't actually mean someone could legally do things like run, modify, distribute, study, or use the code. I highly recommend reading this: https://choosealicense.com/no-permission/

If you would rather keep it closed source (or proprietary), then I don't understand why you have this repository containing the code.

> Please don't open spam issues.

I made the same issue for every repository that did not contain a license declared somewhere (code, repo, readme). Unfortunately, there is a lot of them.

**Comment by Diggsey** [comment]
Yes I know what not having a license means. I fail to see why you would open an issue about it unless you wish to use the code in a way which requires a license - if you do have such a request then we can discuss that.

> I made the same issue for every repository that did not contain a license declared somewhere (code, repo, readme). Unfortunately, there is a lot of them.

There's no requirement to have a license. I published the code on github because it's a convenient way to store and share code. That means the only rights you have to the code are those implied by it being public. What those rights are precisely will depend on the jurisdiction but will generally include the ability to read the code.

What you're doing is just spam unless you have a specific need for a license to use the code.

**Comment by JacksonChen666** [comment]
(Disclaimer: I am not a lawyer, this is not legal advice, and the information provided here might be inaccurate or outdated.)

> Yes I know what not having a license means.

I'm interested in knowing what you think it means to not have a license in a repo.

> I fail to see why you would open an issue about [not having a license] unless you wish to use the code in a way which requires a license

An open source license grants anyone to *explicitly* permit usage, modification, distribution, and studying of the code.

This repo doesn't have an open source license declared somewhere. So those 4 things can't be done safely in a legal sense (because copyright).

> > I made the same issue for every repository that did not contain a license declared somewhere (code, repo, readme). Unfortunately, there is a lot of them.
> 
> There's no requirement to have a license.

That is true, but you [really, really, *really* should add a license](https://choosealicense.com/no-permission/) so people can use your code.

> I published the code on github because it's a convenient way to store and share code.

I don't think it's a problem to use github as a dumping ground for code (dismissing the [major issues with github](https://sfconservancy.org/GiveUpGitHub/)). Even [this project](https://github.com/carykh/jumpcutter) is just a dump of code, but it *actually has a license*.

> That means the only rights you have to the code are those implied by it being public. What those rights are precisely will depend on the jurisdiction but will generally include the ability to read the code.

Maybe that is enough for you, but that completely throws running, modifying, sharing, etc. the code out of the window for everyone else.

> What you're doing is just spam unless you have a specific need for a license to use the code.

I don't have a need for this project. However, someone else who I probably don't know, haven't met or heard about, or will never know about, might need it for one reason or another, but must first resort to asking the maintainer to "Please add a license" so that they can do their things without being posed in a legal risk. Or they'll have to find another project that [did include a license](https://gitlab.com/bpaassen/five_clique).

Here's [another project of yours](https://github.com/Diggsey/query_interface). It has 65 stars, used by [616 repositories](https://github.com/Diggsey/query_interface/network/dependents), and it still does not have a license. (then i link [this](https://choosealicense.com/no-permission/) and tell you to ***read it please***, but there's no better sentence i could think of so i just wrote this instead)

Sourcehut, a code forge service website whatever, tells the owner of a project to add a license, as not having a license may mean that other people may not use the project.

![a view of a test repository on the git service part of sourcehut, with a red box saying "Heads up! We couldn't find a license file for your repository, which means that it may not be possible for others to use this project. If you're unsure of how to choose a license, consult our wiki page, choosing a software license."](https://user-images.githubusercontent.com/50755746/204880789-c95bb0d7-0ccc-4011-9e93-12f0a785e399.png)

<sup>don't tell me to add a license to a git repository containing a file with the name "testa" with nothing in the file. it is completely worthless and the repo is extremely easy to recreate from scratch anyways.</sup>

The reason why I made a wave of issues like this is because [the repositories has been promoted in a video's description](https://youtube.com/watch?v=c33AZBnRHks), and some other reasons written above.

---

tl;dr: Please add a license, so that someone can freely use your code without having to ask you to add a license first.
